src/app/boot/app_controller.nim

```diff
@@ -609,6 +612,17 @@ proc doKeycardReplacement(self: AppController) =
   self.keycardService.startStoreMetadataFlow(keypair.name, self.startupModule.getPin(), accountsPathsToStore)
   info "keycard replacement fully done"
 
+proc runPostOnboardingTasks(self: AppController) =
+    debug "running post-onboarding tasks"
+
+    let tasks = self.onboardingModule.getPostOnboardingTasks()
+    for task in tasks:
+      case task.kind:
```

Comment:
I wonder if we could rely fully on common interface without any additional enum differentiating type of task. But it's because needed dependencies are not available when the task is created, right?

Reply:
@micieslak There are 2 things.
1. For now the `run()` method requires different arguments for different tasks.
We can make common pre-methods like `setAccountsService`, which can later be used (or not) by all tasks. But this sounded a bit nasty.
2. I tried to have `case/of` by the type of the object, but it's not supported in Nim.

So at the end of the day, this solution is straight-forward and it works. I don't like it, but decided not to bother too much.
And It is quite a small thing, we can always refactor if we find a better soltuion.